trade-finance-manager-api/src/graphql/resolvers/query-facilities.js

```diff
@@ -8,17 +10,42 @@ exports.queryAllFacilities = async (queryParams) => {
   const facilities = [];
   // eslint-disable-next-line no-restricted-syntax
   for (const item of rawFacilities) {
-    const facility = item.tfmFacilities;
-    const defaultCoverEndDate = item.tfmFacilities.coverEndDate;
+    const { tfmFacilities: facility } = item;
+    const { facilityId } = facility;
+
+    // checks for amended facility value or coverEndDate
+    const latestCompletedAmendmentValue = await api.getLatestCompletedValueAmendment(facilityId);
```

Comment:
do we need this line and the one below in the loop? if we have over 1000 facilities, that means 2x 1000 API calls

Reply:
fixed so no api calls are made for amendments